Assets/_BForBoss/_Character/Scripts/FirstPersonPlayer.cs

```diff
@@ -66,6 +86,34 @@ protected override void OnStart()
             _wallRunBehaviour?.Initialize(this, base.GetMovementInput, ResetJumpCount);
         }
 
+#if UNITY_EDITOR || DEVELOPMENT_BUILD
+
+        protected override Vector2 GetMouseLookInput()
+        {
+            if (mouseLookInputAction != null && !IsDebugWindowOpen)
+                return mouseLookInputAction.ReadValue<Vector2>();
+
+            return Vector2.zero;
+        }
+
+        protected override void OnCursorLock(InputAction.CallbackContext context)
```

Comment:
do you still need this? since you're using the mouse utility?

Reply:
Nevermind, i understand what this is for now. 

Unfortunately, i can't seem to find a way around this. Best way to handle this is to place this override method within `partial FirstPersonPlayer.Debug class`

And then set the "Debug Flag" on over there.
```
public partial FirstPersonPlayer {
  //override code
}

```

Reply:
OOh I like the Partial Class Idea. Yeah I was scratching my head on this for a while to be honest, really did not want to add this to the FirstPersonPlayer Class!.

Reply:
The reason I would rather use GuiLayout that UGUI is for scalability. The time and effort needed to create UGUI everytime you want to create a Debug Option is a bit much in my opinion. Plus a lot more flexibility with GUI I feel